pkg/s3select/input.go

```diff
@@ -127,15 +130,16 @@ func NewInput(opts *Options) (*Input, error) {
 		tempBytesScanned = opts.StreamSize
 		myReader = bzip2.NewReader(opts.ReadFrom)
 	}
-
+	// DelimiterReader treats custom record delimiter like `\r` and replaces it with `\n`.
+	normalizedReader := ioutil.DelimiterReader(myReader, carriageReturn)
 	progress := &statInfo{
 		BytesScanned:   tempBytesScanned,
 		BytesProcessed: 0,
 		BytesReturned:  0,
 	}
 	reader := &Input{
 		options: opts,
```

Comment:
We shouldn't do this we should use the record delimiter from the XML input for the select API and pass it down as one of the options in SelectInput{} and use it do not use `\r` by default.  This was the whole reason to implement a custom DelimitedReader

https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectSELECTContent.html

RecordDelimiter | The value used to separate individual records in the input. Instead of the default value, you can specify an arbitrary delimiter, including an octal character. For example, \\036 is parsed as the "record separator" (non-printing) character.You can specify up to two characters for a record delimiter. You can specify two characters, one character and one octal, or two octals. For example, \r\n is a valid record delimiter.Type: StringDefault: \nAncestor: CSV
-- | --

Reply:
Done